packages/ragbits-core/src/ragbits/core/vector_stores/base.py

```diff
@@ -31,16 +32,20 @@ class VectorStoreOptions(BaseModel, ABC):
     max_distance: float | None = None
 
 
-class VectorStore(WithConstructionConfig, ABC):
+VectorStoreOptionsType = TypeVar("VectorStoreOptionsType", bound=VectorStoreOptions)
+
+
+class VectorStore(Generic[VectorStoreOptionsType], WithConstructionConfig, ABC):
```

Comment:
I wonder if we can abstract this generic class and create something like a `WithOptions` parent class that takes care of the templates and initialization of `default_options`.

Reply:
Hmm, I don't see a nice way to do it. Also, the initialization of `default_options` happens in `__init__` and is a one-liner. I don't think such simple code needs extra abstraction.

Reply:
It'd make the interface more consistent. Right now, vector stores and embedders handle options differently - one uses class attributes, the other doesn’t - and it feels pretty confusing.